src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -461,10 +461,14 @@ private String getPrincipalName(String username, String domainName) {
                             throw e;
                         }
                     }
-                    if (!found || duration >= 10) {
-                        LOGGER.warning(String.format("AD chain lookup is taking too long (%dms). Falling back to recursive lookup", duration));
+                    if (!found && duration >= 10) {
+                        LOGGER.warning(String.format("AD chain didn't find the group in (%dms). Falling back to recursive lookup", duration));
                         groupLookupStrategy = GroupLookupStrategy.RECURSIVE;
                         continue;
+                    } else if (found && duration >= 10) {
+                        LOGGER.warning(String.format("AD chain lookup found the group but took too long (%dms). Falling back to recursive lookup", duration));
+                        groupLookupStrategy = GroupLookupStrategy.RECURSIVE;
+                        return groups;
                     } else {
                         // it run fast enough, so let's stick to it
```

Comment:
what if `(!found && duration <=10)`, looks like you go in the `else` and return `groups` even if we did not find them